ingest/dense.py

```diff
@@ -33,17 +33,40 @@ def __init__(self, file_path, study_file_id, study_id, **kwargs):
             self, file_path, allowed_file_types=self.ALLOWED_FILE_TYPES
         )
         self.matrix_params = kwargs
+        self.gene_names = []
+        csv_file, self.open_file_object = self.open_file(self.file_path)
+        self.header = next(csv_file)
+
+    def validate_unique_header(self):
+        if len(set(self.header)) != len(self.header):
+            self.error_logger.error(
+                "Duplicated header values are not allowed", extra=self.extra_log_params
+            )
+            return False
+        return True
 
-        self.preprocess()
+    def validate_gene_keyword(self):
+        # File is an R formatted file
+        if (self.header[-1] == '') and (self.header[0].upper() != 'GENE'):
+            pass
+        else:
+            # If not R formatted file then first cell must be 'GENE'
+            if self.header[0].upper() != 'GENE':
+                self.error_logger.error(
```

Comment:
Will this logged error message get passed to the user in the ingest notification email?

Reply:
All error log statements with get written to the same file and are passed to users.